client_cert_test.go

```diff
@@ -0,0 +1,200 @@
+NEhy3N/MSku+RK48njZzb026GyMrjwKrOTTnA81vsljBk9WpZoW2vyBRIStpSlmi
+W2o6eHJzHMilGW4E5+tH8LCCbQZxsh+7qOqliwsHfPCwAlwbHafzphwbYFk2BX6d
+Tt7LbsX+08OzbJltRTNBwbaV8nssKxXQ7ZcbuLmdAoIBAQCZ5LyXn7dDmkcp8jUc
+Tlt8wtbSJNUMe3AQRK5Sl1/cCnaMN8GE5JV7Q6Tocikpm/287flnLUyk0jmIbJcv
+Nb4/kWxpADfsRxLu/458DivPVXnAWb3oBCf4j9HZZNQILRJLgg8YFcDwep85fpn6
+U43zxT5D6If67Wor98yeF3IfO8K2L+n93QvvLq6jx9wCFWCMHqzMFN3HkhhIliCZ
+LUTL/NsI8l+C3oZATt+uLcrccHK6DS7KJ0tcMjO9CCseLBGH4oUrXvRZVoqmCsuU
+7KoKucTiz32rUgwqRW75ijjolYeQxrFTF5AronUFci6c9tnoHpVHyv0MR5ozqfT+
+/fpVAoIBAF+3LwtbXmBvcWs5kRu4P022r0B29O3wZXgG89ubuEVk00E2Ckh9nJf1
+dxb2Ga1GC59xaY8FQxfndSy8VsPFHdt4Vi74bZInXRM4V+SoVY/gN76DGxtQwXjw
+HW4mI3HCWMmn2o8S7/YzbDQFbPTVaDViYlUQ5oDn9QJEumq+zphWOsC4Il3bWm4g
+G435ZYaeQCDmmu4WkQIpKACgZRDQBUs3QdrosprXcCokGr/DJxdjq6dEzoYXCDu3
+Oyf+yXiXi/cLCAe0xwx8zmwbDmC8yPOpj58ETzl/RVh89DEYWr/G1F7sf+8k3nz3
+EJMN4Ww1GVMYvIfdK7cSpDU04LWOKX0=
+-----END PRIVATE KEY-----
+`)
+	testCert = []byte(`-----BEGIN CERTIFICATE-----
+MIIFXTCCA0WgAwIBAgIJAMMAnp6cISAqMA0GCSqGSIb3DQEBCwUAMEUxCzAJBgNV
+BAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEwHwYDVQQKDBhJbnRlcm5ldCBX
+aWRnaXRzIFB0eSBMdGQwHhcNMTYxMDI4MTIwMDUxWhcNMTcxMDI4MTIwMDUxWjBF
+MQswCQYDVQQGEwJBVTETMBEGA1UECAwKU29tZS1TdGF0ZTEhMB8GA1UECgwYSW50
+ZXJuZXQgV2lkZ2l0cyBQdHkgTHRkMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIIC
+CgKCAgEAx15O6ivZlsJjbadoKDbomEjrlt8s5RqobnIMOhJ6+TkIqgmPVdrD1g2V
+HwUhWeiojT9k2TYjPsEHGRLUADfjVfHDp8ZIn6we/UQoYzwjpiIl4kpVeQlYK6OA
+Ly9xfMKO1nB6kIKatB1Xr4C/K+pHoVrSXrtiKRJZLU9m2Ff4RhxpbjzV5aK3gPXU
+EkKw/plzmRryKza2ZTiedgxikESupuvsHpOcJL/+QfExQp01BNcVOgUm1fH1qWls
+FgBt1JRlGeKUU3RtPp4h1eKTXPBtqW+m6/f1ElH2ivfFWHw8RH6i1ideR7an3wJr
+4h6S1BBPK0CL/Aicq0eC1Mx8QjEaQW6Kdgk691PNi7yXqWinvZnjg2Z/ZnB/rRST
+CPCmNSFqb+FCazFHTs++9LQiQ2IJecjRA5IKicwKetl2L4aFek28LfzNwVxjlq9A
+xiCvfytks4sXTkbXlNWh1DrndOEo85H4GS+Z13qbkr7l55EYh6KsdQH9RltbH/td
+sH/xXdudlW+9X0HTGq9iEc2RX+wiii/GWuaOsKphgSHPA8UeA0HzghX9lLT9urGk
+zWbltcNLAxs2pXOhPXpOqcPpISEdF+y/TmOV1lJErcQGGkZKQXNpwA+nu5Dwie2n
+U6MqV+faurW+8buH2WO5uWDJbvEds0y62DWdd8XXx4YJ/zK6Z1kCAwEAAaNQME4w
+HQYDVR0OBBYEFBB2lPL2uqy9z4xvVV6bla1AePuJMB8GA1UdIwQYMBaAFBB2lPL2
+uqy9z4xvVV6bla1AePuJMAwGA1UdEwQFMAMBAf8wDQYJKoZIhvcNAQELBQADggIB
+AG5acpIl15s3u9JpPRShwXGb2ti3o1jbx3riKJAY8jJZTSxQeFARO4GQL+C0Z8+j
+arleCxDZoqq+xDgVITCBHZzkyp9wLbihi8EXMYPbrY0Ncv9IoE5A0Oh2c4iRlQAZ
+KJy2GNgq8aOKzVo4E44aEPfuzgqaGFhX5SBCBanr+cddKQmdsj/XMd+3FGPTZPkZ
+LhG/G7o+tkhewBcd7ZeTSmmYSPTnD6U+12N/UDwzcJoSfNx65j2S8BYhsXv4AnK+
+k1HHqWZEDLW6F8oG4RZb9o1sKG3g65vho0HdZXapPVJ00wczXHAD1JyBm5xZwokM
+MJUT7L+zBktMY4O0YsjiKc9WfeqcnSUF4wQ5p9+DM1eawYTdN1cpmOt7DYohUVcu
+fbgHV2xUJJr3cmvgctEukyp1vbVA2XSQW+cRVZCOxfeC/OKdRR9G76cbM5175DMj
+1CSs8M5pxMGl/arUKrEULA3/9QVbd2BBhg4uJ0DdvczqgYG9PgaoLvb2Ej9Rv/F5
+kGf1sKdsBbTpVZWPLWXsBOamoSu1ufm6AYmlTdRqrvn9QeK19MNw12YFkFq3lwpN
+UmsdOQnlJKUaLwGg3l6/qdZ+YM8S4lYH4ISR8PbknQ8MukUyEmRypjBckFuoMMhW
+GOJqhwmTIXWxyg2JXftsm2geg5GHjFXjcZ5dhkrGegrR
+-----END CERTIFICATE-----`)
+)
+
+func TestGenerateTLSConfigSuccess(t *testing.T) {
```

Comment:
These tests are a bit worrying.
With all others improvements, that I suggested you can probably improve and simplify them.
Also, tests related to `config` should go into `config_test.go`, tests related to `bombardier` into `bombardier_test.go` and so on.